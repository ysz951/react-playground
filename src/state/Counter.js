import React from 'react';
class Counter extends React.Component {
    static defaultProps = {
        step: 1
      };
    state = { count: 0 }
    // constructor(props) {
    //     console.log('props in constructor', props)
    //     super(props)
    //     this.state = { count: 0 }
    //   }
    handleButtonClick() {
        console.log('props in handleButtonClick', this.props)
        console.log('state in handleButtonClick', this.state)
        this.setState({ count: this.state.count + this.props.step })
    }

      // ...
    
    render() {
        return (
            <div>
              <p>The current count: {this.state.count}</p>
              <button onClick= {() => this.handleButtonClick()}>
                Add {this.props.step}
              </button>
            </div>
          )
    }
}

export default Counter