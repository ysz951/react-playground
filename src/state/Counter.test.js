import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Counter from './Counter'

describe(`TheDate Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Counter />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    expect(
      renderer.create(<Counter count={123} step = {5}/>).toJSON()
    ).toMatchSnapshot()
  })
})