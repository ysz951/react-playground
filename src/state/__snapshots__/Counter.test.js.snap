// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheDate Component renders the UI as expected 1`] = `
<div>
  <p>
    The current count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Add 1
  </button>
</div>
`;
